xwiki-commons-core/xwiki-commons-classloader/xwiki-commons-classloader-api/src/main/java/org/xwiki/classloader/URIClassLoader.java

```diff
@@ -96,7 +96,7 @@ public URIClassLoader(URI[] uris)
     public URIClassLoader(URI[] uris, URLStreamHandlerFactory handlerFactory)
     {
         super(new URL[0]);
-        this.finder = new URIResourceFinder(uris, handlerFactory);
+        this.finder = new URIResourceFinder(uris.clone(), handlerFactory);
```

Comment:
If you look at URIResourceFinder constructor you will see that this is useless since it does not keep the table.